src/components/Garden/ProposalDetail/ProposalStatus.tsx

```diff
@@ -81,9 +103,11 @@ const ProposalStatus = ({ proposal }) => {
           `}
         />
       ) : (
-        Icon && <Icon />
+        attributes?.Icon !== undefined && <attributes.Icon />
```

Comment:
![image](https://user-images.githubusercontent.com/691510/150652639-83871cbb-ad10-4cef-a0dd-a26c4095e5a0.png)

If Icon is set to Null lead to bugs.

Two ways to solve, add more check against NuLL or strong type on Icon (immo better approach, because we solve the any type too.)

On line 16

```typescript
     Icon?: typeof IconCheck | typeof IconCross
```

And here on line 106 could be
```typescript
attributes && attributes.Icon && <attributes.Icon />
```

What you think?

Reply:
You are right. But if you see the getStatusAttributes, you can see that there is no Icon equal to null. Or is present or it isn't. Regarding the type yes it would be an improvement for sure.

Reply:
You get too type checking on `<attributes.Icon />`

Reply:
Added the types of the Icon 👌